.github/workflows/unit-tests.yml

```diff
@@ -246,7 +246,7 @@ jobs:
           . ~/virtualenvs/pandas-dev/bin/activate
           python -m pip install --no-cache-dir -U pip wheel setuptools meson[ninja]==1.2.1 meson-python==0.13.1
           python -m pip install numpy -Csetup-args="-Dallow-noblas=true"
-          python -m pip install --no-cache-dir versioneer[toml] cython python-dateutil pytest>=7.3.2 pytest-xdist>=3.4.0 hypothesis>=6.84.0
+          python -m pip install --no-cache-dir versioneer[toml] cython==3.0.10 python-dateutil pytest>=7.3.2 pytest-xdist>=3.4.0 hypothesis>=6.84.0
```

Comment:
Can you add a comment somewhere here about why we're pinning Cython? The we can take this out of draft and merge.

Reply:
Sure, I've added the comment in [378ae75](https://github.com/pandas-dev/pandas/pull/61423/commits/378ae75964268bbe3074a25968970a1131ef3278).